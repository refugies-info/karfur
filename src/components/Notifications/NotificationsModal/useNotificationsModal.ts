import AsyncStorage from "@react-native-async-storage/async-storage";
import { useEffect, useState } from "react";
import { useNotificationsStatus } from "../../../hooks/useNotificationsStatus";

const useNotifcationsModal = (): {
  visible: boolean;
  hide: () => void;
} => {
  const [accessGranted] = useNotificationsStatus();
  const [notificationsModalVisible, setNotificationsModalVisible] =
    useState<boolean>(false);

  useEffect(() => {
    const showModal = async () => {
      const keyExists = await AsyncStorage.getItem("notificationsModal");
      if (!accessGranted) {
        if (!keyExists) {
          await AsyncStorage.setItem("notificationsModal", "true");
          setNotificationsModalVisible(true);
        }
      } else if (accessGranted && notificationsModalVisible) {
        setNotificationsModalVisible(false);
      }
    };

    showModal();
  }, [accessGranted]);

  return {
    visible: notificationsModalVisible,
    hide: () => setNotificationsModalVisible(false),
  };
};

export default useNotifcationsModal;
