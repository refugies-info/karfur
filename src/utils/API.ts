import axios from "react-native-axios";
import { getEnvironment } from "../libs/getEnvironment";

const dbURL = getEnvironment().dbUrl;
const siteSecret = getEnvironment().siteSecret;

const apiCaller = axios.create({
  baseURL: dbURL,
  headers: {
    "Content-Type": "application/json",
    "site-secret": siteSecret,
  },
});

export const getLanguages = () => apiCaller.get("/langues/getLanguages");

export const getCitiesFromGoogleAPI = (text: string) =>
  apiCaller.post(
    `https://maps.googleapis.com/maps/api/place/autocomplete/json?input=${text}&language=fr&types=(cities)&components=country:fr&key=${process.env.GOOGLE_API_KEY}`
  );

export const getCityDetailsFromGoogleAPI = (placeId: string) =>
  apiCaller.post(
    `https://maps.googleapis.com/maps/api/place/details/json?place_id=${placeId}&fields=address_component&key=${process.env.GOOGLE_API_KEY}`
  );

export const getPlaceIdFromLocationFromGoogleAPI = (
  longitude: number,
  latitude: number
) =>
  apiCaller.post(
    `https://maps.googleapis.com/maps/api/geocode/json?latlng=${latitude},${longitude}&key=${process.env.GOOGLE_API_KEY}`
  );

export const getContentsForApp = ({
  locale,
  age,
  department,
  frenchLevel,
}: {
  locale: string;
  age: string | null;
  department: string | null;
  frenchLevel: string | null;
}) => {
  const route =
    `/dispositifs/getContentsForApp?locale=${locale}` +
    (age ? `&age=${age}` : "") +
    (department ? `&department=${department}` : "") +
    (frenchLevel ? `&frenchLevel=${frenchLevel}` : "");

  return apiCaller.get(route);
};

export const get_dispositif = (params = {}) =>
  apiCaller.post("/dispositifs/get_dispositif", params);
