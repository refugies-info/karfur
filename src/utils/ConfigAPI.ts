import axios from "react-native-axios";
import Config from "../libs/getEnvironment";

const dbURL = Config.dbUrl;
const siteSecret = Config.siteSecret;
export const apiCaller = axios.create({
  baseURL: dbURL,
  headers: {
    "Content-Type": "application/json",
    "site-secret": siteSecret,
  },
});