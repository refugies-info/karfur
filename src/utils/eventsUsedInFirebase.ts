export const events = {
  REACHED_END_CONTENT: "REACHED_END_CONTENT",
  LANGUAGE_VALIDATE: "LANGUAGE_VALIDATE",
  LOCATION_VALIDATE: "LOCATION_VALIDATE",
};
