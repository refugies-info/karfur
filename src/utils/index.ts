export { default as withProps } from "./withProps";
