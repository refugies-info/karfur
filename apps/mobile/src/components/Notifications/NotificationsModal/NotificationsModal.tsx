import React from "react";
import { ScrollView } from "react-native";
import Modal from "react-native-modal";
import { useSafeAreaInsets } from "react-native-safe-area-context";
import { useTheme } from "styled-components/native";
import useNotificationsModal from "./useNotificationsModal";
import { EnableNotifications } from "../EnableNotifications";

interface Props {
  /**
   * delay in ms before showing the modal for the first time
   */
  delay?: number;
}

const NotificationsModal = (props: Props) => {
  const { visible, hide } = useNotificationsModal(props.delay || 0);
  const insets = useSafeAreaInsets();
  const theme = useTheme();

  return (
    <Modal isVisible={visible}>
      <ScrollView
        contentContainerStyle={{
          marginTop: insets.top + theme.margin * 2,
          borderRadius: theme.radius * 2,
          backgroundColor: "white",
          padding: theme.margin * 3,
        }}
      >
        <EnableNotifications onDismiss={hide} />
      </ScrollView>
    </Modal>
  );
};

export default NotificationsModal;
