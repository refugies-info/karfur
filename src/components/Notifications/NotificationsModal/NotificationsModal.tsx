import React from "react";
import { ScrollView } from "react-native";
import Modal from "react-native-modal";

import useNotificationsModal from "./useNotificationsModal";
import { EnableNotifications } from "../EnableNotifications";

interface Props {
  /**
   * delay in ms before showing the modal for the first time
   */
  delay?: number;
}

const NotificationsModal = (props: Props) => {
  const { visible, hide } = useNotificationsModal(props.delay || 0);
  return (
    <Modal isVisible={visible}>
      <ScrollView>
        <EnableNotifications onDismiss={hide} />
      </ScrollView>
    </Modal>
  );
};

export default NotificationsModal;
