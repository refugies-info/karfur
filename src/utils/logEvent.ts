/* eslint-disable no-console */
import * as Analytics from "expo-firebase-analytics";
import { getEnvironment } from "../libs/getEnvironment";
import { FirebaseEvent } from "./eventsUsedInFirebase";

export const logEventInFirebase = async (
  eventName: FirebaseEvent,
  data: Record<string, any>
) => {
  const { envName, debugModeFirebase } = getEnvironment();
  if (envName === "DEVELOPMENT") {
    if (debugModeFirebase === "activated") {
      Analytics.setDebugModeEnabled(true);
      console.log(
        "Environment is dev and debug mode is activated for firebase"
      );
    } else {
      console.log(
        "Environment is dev and debug mode is NOT activated for firebase"
      );
      return;
    }
  }

  if (envName && ["DEVELOPMENT", "STAGING", "PROD"].includes(envName)) {
    await Analytics.logEvent(eventName, { ...data, envName });
  }
};
