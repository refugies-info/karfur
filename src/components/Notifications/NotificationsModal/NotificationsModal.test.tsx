import React from "react";
import NotificationsModal from "./NotificationsModal";
import { render } from "../../utils/tests";
import useNotificationsModal from "./useNotificationsModal";

// @see https://stackoverflow.com/questions/50793885/referenceerror-you-are-trying-to-import-a-file-after-the-jest-environment-has
jest.useFakeTimers();

jest.mock("./useNotificationsModal", () => jest.fn());

describe("NotificationsModal snapshot test suite", () => {
  it("should render without bug", async () => {
    (useNotificationsModal as jest.Mock).mockReturnValueOnce({
      visible: true,
      hide: jest.fn(),
    });
    const test = await render(<NotificationsModal />);
    expect(test).toMatchSnapshot();
  });
  it("should render not visible", async () => {
    (useNotificationsModal as jest.Mock).mockReturnValueOnce({
      visible: false,
      hide: jest.fn(),
    });
    const test = await render(<NotificationsModal />);
    expect(test).toMatchSnapshot();
  });
});
