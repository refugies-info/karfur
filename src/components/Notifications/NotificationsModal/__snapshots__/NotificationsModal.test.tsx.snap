// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationsModal snapshot test suite should render not visible 1`] = `null`;

exports[`NotificationsModal snapshot test suite should render without bug 1`] = `
<Modal
  animationType="none"
  deviceHeight={null}
  deviceWidth={null}
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onBackdropPress={[Function]}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  panResponderThreshold={4}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  statusBarTranslucent={false}
  supportedOrientations={
    [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={true}
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    collapsable={false}
    deviceHeight={null}
    deviceWidth={null}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    panResponderThreshold={4}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    statusBarTranslucent={false}
    style={
      {
        "flex": 1,
        "justifyContent": "center",
        "margin": 37.5,
        "transform": [
          {
            "translateY": 1334,
          },
        ],
      }
    }
    supportedOrientations={
      [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <RCTScrollView
      contentContainerStyle={
        {
          "backgroundColor": "white",
          "borderRadius": 12,
          "marginTop": 16,
          "padding": 24,
        }
      }
    >
      <View>
        <View>
          <View
            style={
              {
                "alignItems": "center",
                "marginVertical": "20%",
              }
            }
          >
            <View
              style={
                {
                  "height": 120,
                  "width": 120,
                }
              }
            >
              <LottieAnimationView
                autoPlay={true}
                cacheComposition={true}
                colorFilters={
                  [
                    {
                      "color": 4291348680,
                      "keypath": "bell icon",
                    },
                  ]
                }
                enableMergePathsAndroidForKitKatAndAbove={false}
                loop={true}
                onAnimationFailure={[Function]}
                onAnimationFinish={[Function]}
                onAnimationLoaded={[Function]}
                progress={0}
                resizeMode="cover"
                sourceJson="{"nm":"bell","ddd":0,"h":30,"w":30,"meta":{"g":"@lottiefiles/toolkit-js 0.33.2"},"layers":[{"ty":4,"nm":"bell2 Outlines","sr":1,"st":0,"op":28,"ip":0,"hd":false,"ddd":0,"bm":0,"hasMask":false,"ao":0,"ks":{"a":{"a":0,"k":[14.93,4.945,0],"ix":1},"s":{"a":0,"k":[100,100,100],"ix":6},"sk":{"a":0,"k":0},"p":{"a":0,"k":[14.93,4.945,0],"ix":2},"r":{"a":1,"k":[{"o":{"x":0.333,"y":0},"i":{"x":0.667,"y":1},"s":[0],"t":3},{"o":{"x":0.333,"y":0},"i":{"x":0.667,"y":1},"s":[11],"t":8},{"o":{"x":0.333,"y":0},"i":{"x":0.667,"y":1},"s":[-12],"t":12},{"o":{"x":0.333,"y":0},"i":{"x":0.667,"y":1},"s":[5],"t":16},{"o":{"x":0.333,"y":0},"i":{"x":0.667,"y":1},"s":[-7],"t":19},{"o":{"x":0.333,"y":0},"i":{"x":0.667,"y":1},"s":[3],"t":22},{"s":[0],"t":25}],"ix":10},"sa":{"a":0,"k":0},"o":{"a":0,"k":100,"ix":11}},"ef":[],"shapes":[{"ty":"gr","bm":0,"hd":false,"mn":"ADBE Vector Group","nm":"Group 1","ix":1,"cix":2,"np":2,"it":[{"ty":"sh","bm":0,"hd":false,"mn":"ADBE Vector Shape - Group","nm":"Path 1","ix":1,"d":1,"ks":{"a":0,"k":{"c":true,"i":[[0.553,0],[0,0],[0,-0.553],[-0.552,0],[0,0],[0,0.553]],"o":[[0,0],[-0.552,0],[0,0.553],[0,0],[0.553,0],[0,-0.553]],"v":[[10,-1],[-10,-1],[-11,0],[-10,1],[10,1],[11,0]]},"ix":2}},{"ty":"fl","bm":0,"hd":false,"mn":"ADBE Vector Graphic - Fill","nm":"Fill 1","c":{"a":0,"k":[1,0.7922,0],"ix":4},"r":1,"o":{"a":0,"k":100,"ix":5}},{"ty":"tr","a":{"a":0,"k":[0,0],"ix":1},"s":{"a":0,"k":[100,100],"ix":3},"sk":{"a":0,"k":0,"ix":4},"p":{"a":0,"k":[15,23],"ix":2},"r":{"a":0,"k":0,"ix":6},"sa":{"a":0,"k":0,"ix":5},"o":{"a":0,"k":100,"ix":7}}]},{"ty":"gr","bm":0,"hd":false,"mn":"ADBE Vector Group","nm":"Group 2","ix":2,"cix":2,"np":2,"it":[{"ty":"sh","bm":0,"hd":false,"mn":"ADBE Vector Shape - Group","nm":"Path 1","ix":1,"d":1,"ks":{"a":0,"k":{"c":true,"i":[[0,1.59],[0,0],[3.448,0.89],[0,0],[1.105,0],[0,-1.105],[0,0],[0,-3.726],[0,0],[0.609,-1.453],[0,0]],"o":[[0,0],[0,-3.726],[0,0],[0,-1.105],[-1.105,0],[0,0],[-3.448,0.89],[0,0],[0,1.59],[0,0],[-0.609,-1.452]],"v":[[8,3.894],[8,1.5],[2,-6.238],[2,-6.5],[0,-8.5],[-2,-6.5],[-2,-6.238],[-8,1.5],[-8,3.894],[-8.935,8.5],[8.935,8.5]]},"ix":2}},{"ty":"fl","bm":0,"hd":false,"mn":"ADBE Vector Graphic - Fill","nm":"Fill 1","c":{"a":0,"k":[1,0.7922,0],"ix":4},"r":1,"o":{"a":0,"k":100,"ix":5}},{"ty":"tr","a":{"a":0,"k":[0,0],"ix":1},"s":{"a":0,"k":[100,100],"ix":3},"sk":{"a":0,"k":0,"ix":4},"p":{"a":0,"k":[15,11.5],"ix":2},"r":{"a":0,"k":0,"ix":6},"sa":{"a":0,"k":0,"ix":5},"o":{"a":0,"k":100,"ix":7}}]}],"ind":1},{"ty":4,"nm":"mask","sr":1,"st":0,"op":28,"ip":0,"hd":false,"ddd":0,"bm":0,"hasMask":false,"td":1,"ao":0,"ks":{"a":{"a":0,"k":[14.938,5,0],"ix":1},"s":{"a":0,"k":[100,100,100],"ix":6},"sk":{"a":0,"k":0},"p":{"a":0,"k":[14.938,5,0],"ix":2},"r":{"a":1,"k":[{"o":{"x":0.333,"y":0},"i":{"x":0.667,"y":1},"s":[0],"t":3},{"o":{"x":0.333,"y":0},"i":{"x":0.667,"y":1},"s":[11],"t":8},{"o":{"x":0.333,"y":0},"i":{"x":0.667,"y":1},"s":[-12],"t":12},{"o":{"x":0.333,"y":0},"i":{"x":0.667,"y":1},"s":[5],"t":16},{"o":{"x":0.333,"y":0},"i":{"x":0.667,"y":1},"s":[-7],"t":19},{"o":{"x":0.333,"y":0},"i":{"x":0.667,"y":1},"s":[3],"t":22},{"s":[0],"t":25}],"ix":10},"sa":{"a":0,"k":0},"o":{"a":0,"k":100,"ix":11}},"ef":[],"shapes":[{"ty":"gr","bm":0,"hd":false,"mn":"ADBE Vector Group","nm":"Group 1","ix":1,"cix":2,"np":2,"it":[{"ty":"sh","bm":0,"hd":false,"mn":"ADBE Vector Shape - Group","nm":"Path 1","ix":1,"d":1,"ks":{"a":0,"k":{"c":true,"i":[[0.553,0],[0,0],[0,1.585],[0,0],[4.418,0],[0,-4.418],[0,0],[1.15,-1.893],[0,0],[0,-0.553],[-0.552,0],[0,0],[0,0.553]],"o":[[0,0],[-1.518,-2.129],[0,0],[0,-4.418],[-4.418,0],[0,0],[0,1.585],[0,0],[-0.552,0],[0,0.553],[0,0],[0.553,0],[0,-0.553]],"v":[[10,7.5],[9.923,7.5],[8,0.894],[8,-1.5],[0,-9.5],[-8,-1.5],[-8,0.894],[-9.925,7.5],[-10,7.5],[-11,8.5],[-9.938,9.313],[10.063,9.313],[11,8.5]]},"ix":2}},{"ty":"fl","bm":0,"hd":false,"mn":"ADBE Vector Graphic - Fill","nm":"Fill 1","c":{"a":0,"k":[1,0.7922,0],"ix":4},"r":1,"o":{"a":0,"k":100,"ix":5}},{"ty":"tr","a":{"a":0,"k":[0,0],"ix":1},"s":{"a":0,"k":[100,100],"ix":3},"sk":{"a":0,"k":0,"ix":4},"p":{"a":0,"k":[15,14.5],"ix":2},"r":{"a":0,"k":0,"ix":6},"sa":{"a":0,"k":0,"ix":5},"o":{"a":0,"k":100,"ix":7}}]}],"ind":2},{"ty":4,"nm":"tongue Outlines","sr":1,"st":0,"op":28,"ip":0,"hd":false,"ddd":0,"bm":0,"tt":2,"hasMask":false,"ao":0,"ks":{"a":{"a":0,"k":[3.25,-16.563,0],"ix":1},"s":{"a":0,"k":[100,100,100],"ix":6},"sk":{"a":0,"k":0},"p":{"a":0,"k":[15,5.188,0],"ix":2},"r":{"a":1,"k":[{"o":{"x":0.333,"y":0},"i":{"x":0.667,"y":1},"s":[0],"t":8},{"o":{"x":0.333,"y":0},"i":{"x":0.667,"y":1},"s":[10],"t":11},{"o":{"x":0.333,"y":0},"i":{"x":0.667,"y":1},"s":[-19],"t":14},{"o":{"x":0.333,"y":0},"i":{"x":0.667,"y":1},"s":[8],"t":17},{"o":{"x":0.333,"y":0},"i":{"x":0.667,"y":1},"s":[-14],"t":20},{"o":{"x":0.333,"y":0},"i":{"x":0.667,"y":1},"s":[6],"t":23},{"s":[0],"t":25}],"ix":10},"sa":{"a":0,"k":0},"o":{"a":0,"k":100,"ix":11}},"ef":[],"shapes":[{"ty":"gr","bm":0,"hd":false,"mn":"ADBE Vector Group","nm":"Group 1","ix":1,"cix":2,"np":2,"it":[{"ty":"sh","bm":0,"hd":false,"mn":"ADBE Vector Shape - Group","nm":"Path 1","ix":1,"d":1,"ks":{"a":0,"k":{"c":true,"i":[[1.657,0],[0,-1.657],[-1.657,0],[0,1.657]],"o":[[-1.657,0],[0,1.657],[1.657,0],[0,-1.657]],"v":[[0,-3],[-3,0],[0,3],[3,0]]},"ix":2}},{"ty":"fl","bm":0,"hd":false,"mn":"ADBE Vector Graphic - Fill","nm":"Fill 1","c":{"a":0,"k":[1,0.7922,0],"ix":4},"r":1,"o":{"a":0,"k":100,"ix":5}},{"ty":"tr","a":{"a":0,"k":[0,0],"ix":1},"s":{"a":0,"k":[100,100],"ix":3},"sk":{"a":0,"k":0,"ix":4},"p":{"a":0,"k":[3.25,3.25],"ix":2},"r":{"a":0,"k":0,"ix":6},"sa":{"a":0,"k":0,"ix":5},"o":{"a":0,"k":100,"ix":7}}]}],"ind":3}],"v":"5.6.5","fr":24,"op":28,"ip":0,"assets":[]}"
                speed={1}
                style={
                  {
                    "height": "100%",
                    "width": "100%",
                  }
                }
                textFiltersAndroid={[]}
                textFiltersIOS={[]}
                useNativeLooping={false}
              />
            </View>
            <Text
              isRTL={false}
              selectable={true}
              style={
                {
                  "color": "#212121",
                  "fontFamily": "marianneBold",
                  "fontSize": 24,
                  "lineHeight": 32,
                  "marginTop": 64,
                  "textAlign": "left",
                }
              }
            >
              <Text
                style={{}}
              >
                notifications.enableNotifications
              </Text>
              <View />
            </Text>
            <Text
              isRTL={false}
              selectable={true}
              style={
                {
                  "color": "#212121",
                  "fontFamily": "marianneReg",
                  "fontSize": 16,
                  "lineHeight": 24,
                  "marginTop": 16,
                  "textAlign": "center",
                }
              }
            >
              <Text
                style={{}}
              >
                notifications.beInformed
              </Text>
              <View />
            </Text>
          </View>
          <View
            accessibilityLabel="notifications.enableNotifications"
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#000091",
                "borderColor": "#000091",
                "borderWidth": 1,
                "maxHeight": 100,
                "opacity": 1,
                "paddingBottom": 10,
                "paddingLeft": 10,
                "paddingRight": 10,
                "paddingTop": 10,
                "width": "100%",
              }
            }
          >
            <View
              RTLBehaviour={true}
              animatedStyle={
                {
                  "value": {},
                }
              }
              collapsable={false}
              horizontalAlign="flex-start"
              style={
                {
                  "alignItems": "flex-start",
                  "flexBasis": "auto",
                  "flexDirection": "row",
                  "flexGrow": 1,
                  "justifyContent": "flex-start",
                }
              }
              verticalAlign="flex-start"
            >
              <View
                style={
                  {
                    "flexBasis": "auto",
                    "flexGrow": 0,
                    "marginRight": 8,
                  }
                }
              >
                <Text
                  color="#FFFFFF"
                  disabled={false}
                  isRTL={false}
                  selectable={true}
                  style={
                    {
                      "color": "#FFFFFF",
                      "fontFamily": "marianneMed",
                      "fontSize": 18,
                      "lineHeight": 28,
                      "textAlign": "left",
                    }
                  }
                >
                  <Text
                    style={{}}
                  >
                    notifications.enableNotifications
                  </Text>
                  <View />
                </Text>
              </View>
              <View
                style={
                  {
                    "flexBasis": "auto",
                    "flexGrow": 0,
                  }
                }
              >
                <View
                  disabled={false}
                  style={{}}
                >
                  <View
                    collapsable={false}
                    style={
                      {
                        "transform": [
                          {
                            "scale": 1,
                          },
                        ],
                      }
                    }
                  >
                    <RNSVGSvgView
                      align="xMidYMid"
                      bbHeight="24"
                      bbWidth="24"
                      fill="#FFFFFF"
                      focusable={false}
                      height={24}
                      meetOrSlice={0}
                      minX={0}
                      minY={0}
                      style={
                        [
                          {
                            "backgroundColor": "transparent",
                            "borderWidth": 0,
                          },
                          {
                            "flex": 0,
                            "height": 24,
                            "width": 24,
                          },
                        ]
                      }
                      vbHeight={24}
                      vbWidth={24}
                      width={24}
                    >
                      <RNSVGGroup
                        fill={
                          {
                            "payload": 4294967295,
                            "type": 0,
                          }
                        }
                        propList={
                          [
                            "fill",
                          ]
                        }
                      >
                        <RNSVGGroup
                          fill={
                            {
                              "payload": 4278190080,
                              "type": 0,
                            }
                          }
                        >
                          <RNSVGGroup
                            fill={
                              {
                                "payload": 4278190080,
                                "type": 0,
                              }
                            }
                          >
                            <RNSVGRect
                              fill={
                                {
                                  "payload": 4278190080,
                                  "type": 0,
                                }
                              }
                              height="24"
                              opacity={0}
                              width="24"
                              x="0"
                              y="0"
                            />
                            <RNSVGPath
                              d="M9.86 18a1 1 0 0 1-.73-.32l-4.86-5.17a1 1 0 1 1 1.46-1.37l4.12 4.39 8.41-9.2a1 1 0 1 1 1.48 1.34l-9.14 10a1 1 0 0 1-.73.33z"
                              fill={
                                {
                                  "payload": 4278190080,
                                  "type": 0,
                                }
                              }
                            />
                          </RNSVGGroup>
                        </RNSVGGroup>
                      </RNSVGGroup>
                    </RNSVGSvgView>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            height={16}
            style={
              {
                "minHeight": 16,
              }
            }
          />
          <View
            accessibilityLabel="notifications.notNow"
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderColor": "transparent",
                "borderWidth": 1,
                "maxHeight": 100,
                "opacity": 1,
                "paddingBottom": 10,
                "paddingLeft": 10,
                "paddingRight": 10,
                "paddingTop": 10,
                "width": "100%",
              }
            }
          >
            <View
              RTLBehaviour={true}
              animatedStyle={
                {
                  "value": {},
                }
              }
              collapsable={false}
              horizontalAlign="flex-start"
              style={
                {
                  "alignItems": "flex-start",
                  "flexBasis": "auto",
                  "flexDirection": "row",
                  "flexGrow": 1,
                  "justifyContent": "flex-start",
                }
              }
              verticalAlign="flex-start"
            >
              <View
                style={
                  {
                    "flexBasis": "auto",
                    "flexGrow": 0,
                    "marginRight": 0,
                  }
                }
              >
                <Text
                  color="#000091"
                  disabled={false}
                  isRTL={false}
                  selectable={true}
                  style={
                    {
                      "color": "#000091",
                      "fontFamily": "marianneMed",
                      "fontSize": 18,
                      "lineHeight": 28,
                      "textAlign": "left",
                    }
                  }
                >
                  <Text
                    style={{}}
                  >
                    notifications.notNow
                  </Text>
                  <View />
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</Modal>
`;
